Error: Could not authenticate to the AWS service when setting an empty value to config

```
Updating NEXT_PUBLIC_VERSION
defang config -f compose.portal.dev.gnafed.click.yaml set -e NEXT_PUBLIC_VERSION
 - Reading access token from file /home/runner/.local/state/defang/fabric-prod1.defang.dev
 - Using tenant default for cluster fabric-prod1.defang.dev:443
 - Fabric: v0.6.0-307-gb632a405 CLI: 1.0.6 CLI-Min: v1.0.0
 - tracking event "CONFIG-CREATE INVOKED": [{args [NEXT_PUBLIC_VERSION]} {err <nil>} {non-interactive true} {provider aws} {CalledAs   set} {version 1.0.6} {env true} {file [compose.portal.dev.gnafed.click.yaml]}]
 * Using AWS provider from environment variable
 - Creating aws provider
 - Setting config "NEXT_PUBLIC_VERSION" in project "portal-dev"
 - Putting parameter "/Defang/portal-dev/beta/NEXT_PUBLIC_VERSION"
 - AWS error: operation error SSM: PutParameter, https response error StatusCode: 400, RequestID: 1dd3d715-213f-45e3-bdff-efa0fdfbe17f, api error ValidationException: 1 validation error detected: Value at 'value' failed to satisfy constraint: Member must have length greater than or equal to 1
Error: Could not authenticate to the AWS service. Please check your AWS credentials and try again. (https://docs.defang.io/docs/providers/aws/#getting-started)
Error: Process completed with exit code 2.
```

--- a/src/pkg/cli/client/byoc/aws/byoc.go
+++ b/src/pkg/cli/client/byoc/aws/byoc.go
@@ -105,7 +105,11 @@ func AnnotateAwsError(err error) error {
 		return connect.NewError(connect.CodeNotFound, err)
 	}
 	if cerr := new(smithy.OperationError); errors.As(err, &cerr) {
-		return ErrMissingAwsCreds{err}
+		// Auth can fail for many reasons: ec2imds not available, timeout, no profile, etc. so only check for top-level STS error
+		if cerr.ServiceID == "STS" {
+			return ErrMissingAwsCreds{cerr}
+		}
+		return cerr.Err
 	}
 	if cerr := new(cwTypes.SessionStreamingException); errors.As(err, &cerr) {
 		return connect.NewError(connect.CodeInternal, err)

